dictcc.el

```diff
@@ -271,11 +282,23 @@ At the moment they are of the form `<tr id='trXXX'></tr>'."
                                (dictcc--insert-candidate #'cdr))))))
     (helm :prompt "Filter: " :sources (list source))))
 
+(defun dictcc--select-sourcelang-helm ()
+  "Select source language with helm."
+  (let* ((source `((name . ,"Languages" )
+		   (candidates . ,dictcc-source-languages-alist)
+		   (action . ,(quote identity))
+		   )))
+  (helm :prompt "Filter: " :sources (list source))))
+
+
 ;;;###autoload
 (defun dictcc (query)
   "Search dict.cc for QUERY and insert a result at point."
   (interactive "sQuery: \n")
-  (dictcc--request query))
+  (let ((asklang current-prefix-arg))
```

Comment:
What do you think about using `C-u C-u` to ask for both source and destination language?

Reply:
Good idea, that certainly makes sense. However I am unsure whether my way of reading the prefix argument is the best way possible. In the docs I have often found that this is supposed to be done with an escape command in the `interactive ` call, but I tried this and it looked like it was incompatible with the query that we are doing. Therefore, I think before I add this feature, I need to figure out exactly how handling prefixes works. (It looks like I have not completely understood this)

Reply:
I would recommend using the `interactive` form. If you add a second argument like `ask-for-languages` and append a `p` to the interactive specification, emacs will fill the second argument with the prefix argument, see [the documentation](https://www.gnu.org/software/emacs/manual/html_node/elisp/Interactive-Codes.html). Then you can check `ask-for-languages` and decide which language you should ask the user about. To figure out how it works, evaluate the following code in `M-x ielm` and try `M-x interactive-test` with various prefix arguments.
```emacs-lisp
(defun interactive-test (a b) (interactive "sQuery: \np") (message "%s %s" a b))
```